File: Common/Entity/src/main/java/it/ey/dto/Sezione22DTO.java
package it.ey.dto;

import it.ey.entity.campiTecnici.CampiTecnici;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder(toBuilder = true)
public class Sezione22DTO extends CampiTecniciDTO{

    private  Long id;
    private  Long idPiao;
    private String statoSezione;
    private  String  introPerformance;
    private  String  introObiettiviPerformance;
    private  String  introAdempimenti;
    private  String  introPerformanceOrganizzativa;
    private  String descriptionCollegamentoPerformance;
    private  String introPerformanceIndividuale;
    private List<FaseDTO> fase;
    private List<ObbiettivoPerformanceDTO> obbiettiviPerformance;
    private UlterioriInfoDTO ulterioriInfo;
    private List<AllegatoDTO> allegati;
    private List<AdempimentoDTO> adempimenti;

}

File: PIAO_FE/src/app/shared/models/classes/sezione-22-dto.ts
import { AllegatoDTO } from './allegato-dto';
import { UlterioriInfoDTO } from './ulteriori-info-dto';
import { FaseDTO } from './fase-dto';
import { ObbiettivoPerformanceDTO } from './obiettivo-performance-dto';
import { AdempimentoDTO } from './adempimento-dto';

export class Sezione22DTO {
  id?: number;
  idPiao?: number;
  statoSezione?: string;
  introPerformance?: string;
  introObiettiviPerformance?: string;
  introAdempimenti?: string;
  introPerformanceOrganizzativa?: string;
  descriptionCollegamentoPerformance?: string;
  introPerformanceIndividuale?: string;
  fase?: FaseDTO[];
  obbiettiviPerformance?: ObbiettivoPerformanceDTO[];
  adempimenti?: AdempimentoDTO[];
  ulterioriInfo?: UlterioriInfoDTO;
  allegati?: AllegatoDTO[];
}
